File: chat-service/service/db/src/main/java/org/delivery/db/store/StoreRepository.java
/*
package org.delivery.db.store;

import java.util.List;
import java.util.Optional;
import org.delivery.db.store.vo.StoreCategory;
import org.delivery.db.store.vo.StoreStatus;
import org.springframework.data.jpa.repository.JpaRepository;

public interface StoreRepository extends JpaRepository<Store, Long> {

    Optional<Store> findFirstByIdAndStatusOrderByIdDesc(final Long id, final StoreStatus status);

    List<Store> findAllByStatusOrderByIdDesc(final StoreStatus status);

    List<Store> findAllByStatusAndCategoryOrderByStarDesc(final StoreStatus status, final StoreCategory category);

    Optional<Store> findFirstByNameAndStatusOrderByIdDesc(final String name, final StoreStatus status);

}
*/

File: delivery/service/db/src/main/kotlin/org/delivery/db/store/StoreRepository.kt
package org.delivery.db.store

import org.delivery.db.store.vo.StoreCategory
import org.delivery.db.store.vo.StoreStatus
import org.springframework.data.jpa.repository.JpaRepository

interface StoreRepository : JpaRepository<Store, Long> {

    fun findFirstByIdAndStatusOrderByIdDesc(id: Long?, status: StoreStatus?): Store?

    fun findAllByStatusOrderByIdDesc(status: StoreStatus?): List<Store>

    fun findAllByStatusAndCategoryOrderByStarDesc(status: StoreStatus?, category: StoreCategory?): List<Store>

    fun findFirstByNameAndStatusOrderByIdDesc(name: String?, status: StoreStatus?): Store?

}
